pandas/core/indexes/datetimelike.py

```diff
@@ -289,21 +289,17 @@ def max(self, axis=None, skipna=True, *args, **kwargs):
             return self._na_value
 
         i8 = self.asi8
-        try:
-            # quick check
-            if len(i8) and self.is_monotonic and i8[-1] != iNaT:
-                return self._data._box_func(i8[-1])
-
-            if self.hasnans:
-                if skipna:
-                    max_stamp = self[~self._isnan].asi8.max()
-                else:
-                    return self._na_value
-            else:
-                max_stamp = i8.max()
-            return self._data._box_func(max_stamp)
-        except ValueError:
+
+        # quick check
```

Comment:
we _could_ just return `self._data.max(...)`.  Downside is that doesn't take advantage of caching